types/dragino/lht65n-ne117/uplink.js

```diff
@@ -0,0 +1,277 @@
+}
+
+function decoder(bytes, port) {
+  const ext = bytes[6] & 0x0f;
+  const pollMessageStatus = (bytes[6] >> 7) & 0x01;
+  const connect = (bytes[6] & 0x80) >> 7;
+  const decode = { lifecycle: {}, decoded: {}, external: {}, datalog: {} };
+  if (((port === 3) & (bytes[2] === 0x01)) | (bytes[2] === 0x02)) {
+    const array1 = [];
+    const bytes1 = "0x";
+    const str1 = Str1(bytes);
+    const str2 = str1.substring(0, 6);
+    const str3 = str1.substring(6);
+    const reg = /.{4}/g;
+    const rs = str3.match(reg);
+    rs.push(str3.substring(rs.join("").length));
+    rs.pop();
+    const newSet = new Set(rs);
+    const newArr = [...newSet];
+    const data1 = newArr;
+    decode.lifecycle.batteryVoltage = parseInt(
+      (bytes1 + str2.substring(0, 4)) & 0x3fff,
+    );
+    let batteryLevel =
+      Math.round((decode.lifecycle.batteryVoltage - 2.5) / 0.005 / 10) * 10;
+
+    if (batteryLevel > 100) {
+      batteryLevel = 100;
+    } else if (batteryLevel < 0) {
+      batteryLevel = 0;
+    }
+
+    decode.lifecycle.batteryLevel = batteryLevel;
+
+    if (parseInt(bytes1 + str2.substring(4)) === 1) {
+      decode.lifecycle.sensor = "DS18B20";
+    } else {
+      decode.lifecycle.sensor = "TMP117";
+    }
+    for (let i = 0; i < data1.length; i++) {
+      const temp = parseInt(bytes1 + data1[i].substring(0, 4)) / 100;
+      array1[i] = temp;
+    }
+    decode.decoded.temperature = array1;
+    return decode;
+  }
+  switch (pollMessageStatus) {
+    case 0:
+      if (ext === 0x09) {
+        decode.decoded.temperature = parseFloat(
+          ((((bytes[0] << 24) >> 16) | bytes[1]) / 100).toFixed(2),
+        );
+        decode.lifecycle.batteryStatus = bytes[4] >> 6;
+      } else {
+        decode.lifecycle.batteryVoltage =
+          (((bytes[0] << 8) | bytes[1]) & 0x3fff) / 1000;
+        let batteryLevel =
+          Math.round((decode.lifecycle.batteryVoltage - 2.5) / 0.005 / 10) * 10;
+
+        if (batteryLevel > 100) {
+          batteryLevel = 100;
+        } else if (batteryLevel < 0) {
+          batteryLevel = 0;
+        }
+
+        decode.lifecycle.batteryLevel = batteryLevel;
+
+        let batteryStatus = bytes[0] >> 6;
+        switch (batteryStatus) {
+          case 0:
+            batteryStatus = "ULTRA_LOW";
+            break;
+          case 1:
+            batteryStatus = "LOW";
+            break;
+          case 2:
+            batteryStatus = "OK";
+            break;
+          case 3:
+            batteryStatus = "GOOD";
+            break;
+
+          default:
+            break;
+        }
+
+        decode.lifecycle.batteryStatus = batteryStatus;
+      }
+
+      if (ext !== 0x0f) {
+        decode.decoded.temperature = parseFloat(
+          ((((bytes[2] << 24) >> 16) | bytes[3]) / 100).toFixed(2),
+        );
+        decode.decoded.humidity = parseFloat(
+          ((((bytes[4] << 8) | bytes[5]) & 0xfff) / 10).toFixed(1),
+        );
+      }
+      if (connect === "1") {
```

Comment:
make sure the types of these comparisons are correct, it does not seem that connect is a string